4_gap_analysis/src/3_conusMap.R

```diff
@@ -0,0 +1,75 @@
+#setwd("D:/abock/NAWQA/WQR/nawqa_wqp")
+
+# Bugs in tidyverse interfere with ggplot2 maps functionality
+# https://stackoverflow.com/questions/45066628/cannot-run-map-datastate
+# library(tidyverse)
+library(leaflet)
+library(tidyverse)
+library(mapview)
+library(htmlwidgets)
+library(webshot)
+
+#sapply(list.files("4_gap_analysis/src",full.names = T),source,.GlobalEnv)
+siteFiles<-dir('4_gap_analysis/in/Site', pattern='*sites.feather',full.names = T)
+# Files generated by filtering (by site)
+sampFiles<-dir('4_gap_analysis/in/Tiers', pattern='*.feather',full.names = T)
+
+# combined data may not work
+siteInfo <- bind_rows(
+  lapply(siteFiles, function(dfile) {
+    dat <- feather::read_feather(dfile)
+  })
+)
+siteInfo<-siteInfo %>% distinct(MonitoringLocationIdentifier,.keep_all=T)
+
+sampleInfo<- bind_rows(
+  lapply(sampFiles, function(dfile) {
+    dat <- feather::read_feather(dfile)
+  })
+)
+
+TierA<-sampleInfo %>%
+  filter(numYears > 4 & numCon==3 & Quart > 19) %>%
+  mutate(Tier="a) 3 consituents, 5 or more years, all quarters")
+
+TierB<-sampleInfo %>%
+  filter(numYears > 4 & numCon==3 & Quart < 20)%>%
+  mutate(Tier="b) 3 cons., less than quarterly")
+
+TierC<-sampleInfo %>%
+  filter(numYears > 4 & numCon<3 & Quart > 19)%>%
+  mutate(Tier="c) 2 constituents, 5 or more years, all quarters")
```

Comment:
One thing I was wondering during the last monthly phone call: would there be any value to splitting this analysis by constituent? I ask because it seems like trends are the clearest focus question, and those are usually computed for just one constituent at a time, right? I do still agree that it'd be really great to have a network of sites where multiple constituents were measured (partly so the 1-constituent monitors don't get cocky, and partly so others could do analyses of in-stream nutrient ratios)...but we could probably get higher numbers if we were looking for quarterly sampling of just TN, quarterly sampling of just phosphate, etc., couldn't we? And if having a different set of sites for each constituent isn't a huge problem, then we could even think about incentivizing it that way - a provider's sites could get one gold star per constituent that's monitored with high frequency, one silver star per constituent that's monitored with medium frequency, etc.

Reply:
I've developed have duplicate code that looks at a single constituent at a time; just haven't committed to yet.  I will add this for the next update.

Reply:
Cool!